tests/utils.js

```diff
@@ -1,7 +1,214 @@
+  num2hexstring,
+  num2VarInt,
+  reverseArray,
+  reverseHex,
+  sha256,
+  str2ab,
+  str2hexstring
+} from '../src/utils'
 
 describe('Utils', () => {
-  describe('num2VarInt', () => {
+  describe('ab2str', () => {
+    const testingab = [84, 101, 115, 116]
+    it('converts an empty array to an empty string', () => {
+      const actual = ab2str([])
+      const expected = ''
+      actual.should.eql(expected)
+    })
+    it('converts an array of ascii bytes to string', () => {
+      const actual = ab2str(testingab)
+      const expected = 'Test'
+      actual.should.eql(expected)
+    })
+    it('converts an arraybuffer of ascii bytes to string', () => {
+      const buffer = new Uint8Array(testingab).buffer
+      const actual = ab2str(buffer)
+      const expected = 'Test'
+      actual.should.eql(expected)
+    })
+  })
+
+  describe('str2ab', () => {
+    it('throws if non-string', () => {
+      (() => str2ab([0])).should.throw()
+    })
+    it('converts an empty string to an empty array', () => {
+      const actual = str2ab('')
+      actual.length.should.eql(0)
+    })
+    it('converts a string to an array of ascii bytes', () => {
+      const actual = str2ab('Test')
+      const expected = new Uint8Array([84, 101, 115, 116])
+      actual.should.eql(expected)
+    })
+  })
+
+  describe('hexstring2ab', () => {
+    it('throws if non-string', () => {
+      (() => hexstring2ab([0])).should.throw()
+    })
+    it('converts an empty string to an empty array', () => {
+      const actual = hexstring2ab('')
+      actual.length.should.eql(0)
+    })
+    it('converts a hex string to an array of ascii bytes', () => {
+      const actual = hexstring2ab('54657374')
+      actual.should.eql(new Uint8Array([84, 101, 115, 116]))
+    })
+  })
+
+  describe('ab2hexstring', () => {
+    it('throws if non-array', () => {
+      (() => ab2hexstring('Test')).should.throw()
+    })
+    it('converts an empty array to an empty string', () => {
+      const actual = ab2hexstring([])
+      actual.should.eql('')
+    })
+    it('converts a an array of ascii bytes to a hex string', () => {
+      const actual = ab2hexstring(new Uint8Array([84, 101, 115, 116]))
+      actual.should.eql('54657374')
+    })
+  })
+
+  describe('str2hexstring', () => {
+    it('throws if non-string', () => {
+      (() => str2hexstring([0])).should.throw()
+    })
+    it('converts an empty string to an empty string', () => {
+      const actual = str2hexstring('')
+      actual.should.eql('')
+    })
+    it('converts a string to a hex string', () => {
+      const actual = str2hexstring('Test')
+      actual.should.eql('54657374')
+    })
+  })
+
+  describe('int2hex', () => {
+    it('throws if non-number', () => {
+      (() => int2hex('01')).should.throw()
+    })
+    it('converts a number to its hex form', () => {
+      const actual = int2hex(0x00)
+      actual.should.eql('00')
+    })
+    it('converts a number to its hex form (max)', () => {

```

Comment:
This isnt correct. `int2hex` can convert any size. The biggest difference between this and `num2hexstring` is that for `int2hex` you do not need to constraint the size. Thus, `int2hex` is useful for converting arbitrary data but `num2hexstring` requires an upper limit.
```
int2hex(88888) = '015b38' // Doesnt care abt size, just ensures that there is a leading zero if there length is odd.
num2hexstring(88888, 4) = '5b38' // Cares and truncates to length
```